synapse/models/inet.py

```diff
@@ -208,6 +208,23 @@ class Fqdn(s_types.Type):
     def postTypeInit(self):
         self.setNormFunc(str, self._normPyStr)
 
+    def _ctorCmprEq(self, text):
+        if text == '':
```

Comment:
To facilitate the "correct" behavior for an empty string, we should just construct a cmpr function that always returns False to handle a case where the cmpr is being given against a var that has ""

Reply:
The current behavior, prior to this, is to raise BadTypeValu since the empty string fails to normalize. This preserves the exception throwing behavior, as opposed to silently letting things through.